src/mon/OSDMonitor.cc

```diff
@@ -2137,6 +2153,41 @@ int OSDMonitor::get_osd_objectstore_type(int osd, string *type)
   return 0;
 }
 
+void OSDMonitor::get_filestore_osd_list()
+{
+  for (unsigned osd = 0; osd < osdmap.get_num_osds(); ++osd) {
+    string objectstore_type;
+    int r = get_osd_objectstore_type(osd, &objectstore_type);
+    if (r == 0 && objectstore_type == "filestore") {
+      filestore_osds.insert(osd);
+    }
+  }
+}
+
+void OSDMonitor::check_for_filestore_osds(health_check_map_t *checks)
+{
+  ostringstream ss;
+  if (g_conf()->mon_warn_on_filestore_osds &&
+      filestore_osds.size() > 0) {
+    ss << filestore_osds.size()
+       << " osd(s) "
+       << (filestore_osds.size() == 1 ? "is" : "are")
+       << " on filestore; Filestore has been deprecated,"
+       << " the 'osd_op_queue' is enforced to 'wpq' for filestore OSDs.";
+    auto& d = checks->add("OSD_FILESTORE", HEALTH_WARN, ss.str(),
+                          filestore_osds.size());
+    ss.clear();
```

Comment:
FreeBSD still does not have BlueStore....
I have/had patches up as PR, but they are sitting, and are probably bit-roted again.
Would appriciate this check to be excluded on FreeBSD 
    `#if defined(__FreeBSD__)`